controller/controller_server/csi_controller_server.py

```diff
@@ -676,18 +676,20 @@ def _get_volume_final_name(self, request, array_mediator):
         return self._get_object_final_name(request, array_mediator.max_volume_prefix_length,
                                            array_mediator.max_volume_name_length,
                                            config.VOLUME_TYPE_NAME,
-                                           config.PARAMETERS_VOLUME_NAME_PREFIX)
+                                           config.PARAMETERS_VOLUME_NAME_PREFIX,
+                                           array_mediator.default_object_prefix)
 
     def _get_snapshot_final_name(self, request, array_mediator):
         name = self._get_object_final_name(request, array_mediator.max_snapshot_prefix_length,
                                            array_mediator.max_snapshot_name_length,
                                            config.SNAPSHOT_TYPE_NAME,
-                                           config.PARAMETERS_SNAPSHOT_NAME_PREFIX)
+                                           config.PARAMETERS_SNAPSHOT_NAME_PREFIX,
+                                           array_mediator.default_object_prefix)
         return name
 
     def _get_object_final_name(self, request, max_name_prefix_length, max_name_length, object_type,
-                               prefix_param_name):
-        full_name = name = request.name
+                               prefix_param_name, default_prefix):
```

Comment:
I'm starting to think we should pass the mediator, instead of many of its properties

Reply:
the mediator is very big, should we?

Reply:
it's only 1 reference, compared to current 3

Reply:
6, snapshot and volume, how do you want to unite them? (and we really want to have size for each volume type?)

Reply:
perhaps I wasn't clear - I only suggest to pass `mediator` instead of `max_name_prefix_length`, `max_name_length` and `default_prefix`

Reply:
look at the parameters source.. the calling functions not using the same values, how can we know what values to use?

Reply:
current calls:
```
self._get_object_final_name(request, array_mediator.max_volume_prefix_length,
                            array_mediator.max_volume_name_length,
                            config.VOLUME_TYPE_NAME,
                            config.PARAMETERS_VOLUME_NAME_PREFIX,
                            array_mediator.default_object_prefix)

self._get_object_final_name(request, array_mediator.max_snapshot_prefix_length,
                            array_mediator.max_snapshot_name_length,
                            config.SNAPSHOT_TYPE_NAME,
                            config.PARAMETERS_SNAPSHOT_NAME_PREFIX,
                            array_mediator.default_object_prefix)
```
my suggestion:
```
self._get_object_final_name(request, array_mediator,
                            config.VOLUME_TYPE_NAME,
                            config.PARAMETERS_VOLUME_NAME_PREFIX)

self._get_object_final_name(request, array_mediator,
                            config.SNAPSHOT_TYPE_NAME,
                            config.PARAMETERS_SNAPSHOT_NAME_PREFIX)
```
hope it's clear

Reply:
@ArbelNathan , what do you mean by "the mediator is very big"? Is it a "god object"? 
Maybe it's another thing we can improve.
I'm not sure it's relevant for this task though.

Reply:
@oriyarde and inside `_get_object_final_name` should we use `max_volume_name_length` or `max_snapshot_name_length` or add if for every call and choose by `<>_TYPE_NAME`? cause its not sound pretty

Reply:
@zingero Im not sure I understand the term "god object", can you elaborate?

Reply:
God object is an object that knows too much or does too much. The God object is an example of an anti-pattern
https://en.wikipedia.org/wiki/God_object

Reply:
@ArbelNathan I see what you mean.
if `max_volume_name_length` and `max_snapshot_name_length` are always the same, then perhaps we can unify them to one `max_object_name_length`.
if not, then I agree that choosing by `<>_TYPE_NAME` isn't pretty.
feel free to unify them, open a ticket to unify them, or just resolve this conversation without any change.

@zingero regarding the god object - we can move some of the mediator properties values to its abstract class or other objects, such as `Volume` and `Snapshot`, etc...
but it's definitely not for this ticket.

Reply:
Of course, we'll improve it in the future.

Reply:
I opened CSI-2794